feat(gallery): include hidden navigation buttons on touch aware devices

maxfrigge opened the pull request:
This was discussed in #210 

I have implemented the feature to always include navigation buttons in the markup.
On touch aware devices the lightbox instance will have a css class `featherlight-swipe-aware`.
The default styles will then hide the navigation buttons to stay backward compatible.

You can then choose to override the default and display navigation buttons with:

``` css
.featherlight-swipe-aware .featherlight-next,
.featherlight-swipe-aware .featherlight-previous {
  display: inline-block;
}
```

marcandre commented:
Looks pretty good!

Only missing a comment in the ChangeLog.

Also, the `toggleClass` could be simplified with `addClass`, right?

I think not having a test is fine.

maxfrigge commented:
You are absolutely right about `toggleClasse` - missed that after moving the code.
Any suggestion what the change log should mention exactly?

marcandre commented:
How about:

> Navigation can now be shown on mobile. It is now hidden on mobile with CSS, so be sure to update the default CSS as well as the code! [#237]

maxfrigge commented:
That's spot on :)

I have update the PR and also bumped the version number

marcandre commented:
Great. Except I'd like to squash your commits, and would rather do the release myself. It's not clear, but the best way is to update the "master" section of the Changelog. That gets moved when releasing... Anyways, if you can't easily revert the release, I'll cherrypick things, just let me know

ScotsScripts commented:
Is this update going to make it so swiping with the featherlight gallery works on touch screens but at the same time, if a person is using a mouse on a touch screen (like a surface or something), they can still navigate with mouse by clicking the R/L arrows that come up on hover? This would be a great update if so!

marcandre commented:
No, not quite. I'm actually not even sure how to detect mouse capability?

ScotsScripts commented:
Oh, too bad, the problem is when I have an external non-touch monitor connected to my touch enabled Surface and the  browser is in the external monitor, there's no way to use the featherlight gallery because the system seems to think the non-touch external monitor has touch capabilities. I was hoping your update would enable mouse clicks on the gallery regardless if the screen was touch or not. I'm not sure why it was coded to be one or the other there are a windows 10 laptops with touch screens where people use the mouse for navigation unless they are in tablet mode.

marcandre commented:
Might be possible to detect mouse, by listening to `mousemove` and making sure there's not a `mousedown` at the same palce right after (see [this thread](http://stackoverflow.com/questions/7838680/detecting-that-the-browser-has-no-mouse-and-is-touch-only) and its comments).

If that's the case, we could also add a class 'mouse-aware'.

Or maybe it's not even needed, just showing the buttons on `:hover` no matter what should be good enough?

ScotsScripts commented:
Just showing the buttons on hover no matter what would be a great solution for this! Hopefully it's a simple tweak and can be included in the next release.

MaluNoPeleke commented:
Does that mean swiping and clicking will work on all (supported) devices with the next update?
I just found out about Featherlight and I am looking for a clean and small solution that works similar to https://www.computerbase.de/2016-08/samsung-gear-fit2-test/

robneu commented:
Swiping and clicking already work fine if you use on of the libraries mentioned [here](https://github.com/noelboss/featherlight/wiki/Gallery:-swipe-on-touch-devices). 

I don't really understand the point of this PR to be honest. Does this do anything other than add a class for swipe devices? Is that really useful?

ScotsScripts commented:
@robneu Swiping and clicking work, but not at the same time on a touch screen device. If you have one of the swipe libraries installed and are using a windows 10 touch screen notebook/tablet and you're navigating via mouse, the hover-arrows do not show up in the gallery to move to the next image. Clicking and swiping need to be active at the same time on a touch enabled device so the user isn't forced to take their hand off the mouse to swipe to the next image.

robneu commented:
Isn't that just a CSS issue? It's been a long time since I looked at the default featherlight CSS. Our WP plugin version of it just displays gallery arrows on all device sizes because not having them on swipe devices confused users.

robneu commented:
Ah, I see what the PR addresses now. The current gallery just doesn't output the buttons on touch devices at all. Sorry for the confusion.

maxfrigge commented:
@marcandre I have squashed the commits and removed the the release stuff (which included the not in changelog). So please add the changelog info when you create the release.
